lib/excon/error.rb

```diff
@@ -48,9 +48,33 @@ def initialize(socket_error = Excon::Error.new)
 
     class InvalidHeaderKey < Error; end
     class InvalidHeaderValue < Error; end
-    class Timeout < Error; end
     class ResponseParse < Error; end
 
+    class Timeout < Error
+      def self.by_type(type, human_name = type)
+        case type
+        when :connect_read
+          Excon::Errors::ConnectReadTimeout.described_as(human_name)
+        when :connect_write
+          Excon::Errors::ConnectWriteTimeout.described_as(human_name)
+        when :read
+          Excon::Errors::ReadTimeout.described_as(human_name)
+        when :write
+          Excon::Errors::WriteTimeout.described_as(human_name)
+        end
+      end
```

Comment:
Not sure if `by_type` is clear enough. It seems to make sense in caller context.

Reply:
It seems pretty good. The only further thing that comes to mind would be to tweak it toward something like `new_by_type` since it is kind of a specialized constructor that might make it a little more obvious the intention. What do you think?